src/ansi-c/c_typecheck_type.cpp

```diff
@@ -1449,25 +1452,46 @@ void c_typecheck_baset::typecheck_symbol_type(typet &type)
     error() << "expected type symbol" << eom;
     throw 0;
   }
+}
 
-  if(symbol.is_macro)
-  {
-    // overwrite, but preserve (add) any qualifiers and other flags
+void c_typecheck_baset::typecheck_typedef_type(typet &type)
```

Comment:
Why not take a `typedef_typet` to avoid the immediate cast (makes more consistent with `typecheck_symbol_type` as well).

Reply:
The typedef_type is replaced by whatever the typedef points to; i.e., you would violate the type invariant upon returning from this method.